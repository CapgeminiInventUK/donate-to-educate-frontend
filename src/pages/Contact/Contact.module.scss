@use '@assets/global.scss' as g;

.container {
  background: g.$primary-light-blue;
  height: 100%;

  .contentContainer {
    width: 80%;
    margin: auto;
    margin-bottom: 66px;

    h1 {
      padding: 66px 0;
      margin: 0;
    }

    .card {
      border-radius: 10px;
      background: g.$primary-white;
      padding: 40px 82px 60px;
      box-shadow: 0px 4px 4px 0px rgba(0, 0, 0, 0.1);
      display: flex;
      justify-content: center;
      align-items: center;
      flex-direction: column;

      .email {
        word-break: break-word;
        color: g.$primary-mid-blue;
        font-size: 32px;
        font-weight: 700;
        line-height: 125%; /* 40px */
        letter-spacing: -0.32px;
        text-decoration-line: underline;
        margin: 40px 0;
      }

      .home {
        color: g.$primary-dark-blue;
        font-size: 16px;
        font-weight: 500;
        line-height: 160%; /* 25.6px */
        text-decoration-line: underline;
        display: inline-block;
      }
    }
  }
}

@media only screen and (max-width: calc(g.$screen-medium)) {
  .title {
    display: none;
  }

  .card {
    margin-top: 66px;
  }
}
