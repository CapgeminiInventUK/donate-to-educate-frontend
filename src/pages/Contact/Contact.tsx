import DonateToEducateBanner from '@/components/DonateToEducateBanner/DonateToEducateBanner';
import styles from './Contact.module.scss';
import { FC } from 'react';
import Paths from '@/config/paths';
import { Link } from 'react-router-dom';
import Email from '@/assets/tiles/Email';

const Contact: FC = () => {
  return (
    <div className={styles.container}>
      <DonateToEducateBanner />
      <div className={styles.contentContainer}>
        <h1 className={styles.title}>Contact us</h1>
        <div className={styles.card}>
          <Email />
          <Link to={Paths.EMAIL} className={styles.email}>
            team@donatetoeducate.org.uk
          </Link>
          <Link className={styles.home} to={Paths.HOME}>
            Return to homepage
          </Link>
        </div>
      </div>
    </div>
  );
};

export default Contact;
