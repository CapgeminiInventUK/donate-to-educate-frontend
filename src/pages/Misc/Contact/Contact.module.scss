@use '@assets/global.scss' as g;

.container {
  @include g.page-container;

  .contentContainer {
    width: 80%;
    margin: auto;
    margin-bottom: 66px;

    h1 {
      padding: 0 0 26px 0;
      margin: 0;
    }

    .contactCard {
      padding: 40px 82px 60px;
      display: flex;
      justify-content: center;
      align-items: center;
      flex-direction: column;

      .email {
        word-break: break-word;
        color: g.$primary-mid-blue;
        font-size: 32px;
        font-weight: 700;
        line-height: 125%; /* 40px */
        letter-spacing: -0.32px;
        text-decoration-line: underline;
        margin: 40px 0;
      }

      .home {
        color: g.$primary-dark-blue;
        font-size: 16px;
        font-weight: 500;
        line-height: 160%; /* 25.6px */
        text-decoration-line: underline;
        display: inline-block;
      }
    }
  }
}

@media only screen and (max-width: calc(g.$screen-medium)) {
  .container {
    padding: 0px 19px;

    .contentContainer {
      width: 100%;
      padding-top: 20px;

      .contactCard {
        padding: 20px;

        .email {
          font-size: 20px;
        }
      }
    }
  }
}

@media only screen and (max-width: calc(g.$screen-small)) {
  .container {
    .contentContainer {
      width: 100%;
      margin-bottom: 20px;
      padding-top: 20px;

      .contactCard {
        padding: 20px;
        margin: 20px;
        margin-top: 0;

        .email {
          font-size: 16px;
        }
      }
    }
  }
}
