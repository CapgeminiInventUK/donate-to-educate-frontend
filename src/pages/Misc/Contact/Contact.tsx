import styles from './Contact.module.scss';
import { FC } from 'react';
import Paths from '@/config/paths';
import { Link } from 'react-router-dom';
import Email from '@/assets/contact/Email';
import Phone from '@/assets/contact/Phone';
import BackButton from '@/components/BackButton/BackButton';
import Card from '@/components/Card/Card';

const Contact: FC = () => {
  return (
    <div className={styles.container}>
      <div className={styles.contentContainer}>
        <BackButton theme="blue" />
        <h1 className={styles.title}>Contact us</h1>
        <Card className={styles.contactCard}>
          <Email />
          <Link to={Paths.EMAIL} className={styles.email}>
            team@donatetoeducate.org.uk
          </Link>
          <Phone />
          <Link to={Paths.PHONE} className={styles.email}>
            0134 271 8679
          </Link>
          <Link className={styles.home} to={Paths.HOME}>
            Return to homepage
          </Link>
        </Card>
      </div>
    </div>
  );
};

export default Contact;
