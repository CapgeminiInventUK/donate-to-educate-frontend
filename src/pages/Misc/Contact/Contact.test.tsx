import { create } from 'react-test-renderer';

import Contact from './Contact';
import { BrowserRouter as Router } from 'react-router-dom';

it('renders correctly', () => {
  const tree = create(
    <Router>
      <Contact />
    </Router>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
