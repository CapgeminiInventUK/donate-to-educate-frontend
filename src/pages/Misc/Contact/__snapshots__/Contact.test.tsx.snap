// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="contentContainer"
  >
    <button
      className="back  blue"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.3996 16.8008L9.59961 12.0008L14.3996 7.20078"
          stroke="#11356F"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="3"
        />
      </svg>
      <h4>
        Back
      </h4>
    </button>
    <h1
      className="title"
    >
      Contact us
    </h1>
    <div
      className="card"
    >
      <svg
        fill="none"
        height="97"
        viewBox="0 0 98 97"
        width="98"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#4A8FE7"
          height="97"
          rx="48.5"
          width="97"
          x="0.5"
        />
        <path
          d="M25.4873 31.4873L47.1833 46.5076C48.4361 47.3749 50.0952 47.3749 51.3479 46.5076L73.0439 31.4873M27.3164 69.8984H71.2148C75.2556 69.8984 78.5312 66.6228 78.5312 62.582V33.3164C78.5312 29.2757 75.2556 26 71.2148 26H27.3164C23.2757 26 20 29.2757 20 33.3164V62.582C20 66.6228 23.2757 69.8984 27.3164 69.8984Z"
          stroke="#FEFCFD"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="6"
        />
      </svg>
      <a
        className="email"
        href="mailto: team@donatetoeducate.org.uk"
      >
        team@donatetoeducate.org.uk
      </a>
      <svg
        fill="none"
        height="97"
        viewBox="0 0 98 97"
        width="98"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#4A8FE7"
          height="97"
          rx="48.5"
          width="97"
          x="0.5"
        />
        <path
          d="M50.6777 33.8148C53.6774 34.3176 56.4023 35.7344 58.5777 37.9053C60.753 40.0762 62.1613 42.7955 62.6765 45.7891M51.1357 24C56.4711 24.9026 61.337 27.4278 65.2068 31.2783C69.0767 35.1402 71.5955 39.9962 72.5 45.3206M68.3324 66.5042C68.3324 66.5042 65.4357 69.3492 64.7259 70.1833C63.5695 71.4173 62.207 72 60.421 72C60.2492 72 60.066 72 59.8943 71.9886C56.4939 71.7715 53.3339 70.4461 50.9639 69.3149C44.4836 66.1842 38.7933 61.7396 34.0648 56.1066C30.1606 51.4106 27.5502 47.0688 25.8213 42.407C24.7566 39.562 24.3673 37.3454 24.539 35.2545C24.6535 33.9176 25.1687 32.8093 26.119 31.861L30.0232 27.9648C30.5842 27.4392 31.1796 27.1535 31.7635 27.1535C32.4848 27.1535 33.0687 27.5877 33.4351 27.9533C33.4465 27.9648 33.458 27.9762 33.4694 27.9876C34.1678 28.6389 34.8319 29.313 35.5303 30.0328C35.8852 30.3985 36.2516 30.7641 36.618 31.1411L39.7436 34.2604C40.9572 35.4715 40.9572 36.5913 39.7436 37.8024C39.4116 38.1338 39.091 38.4651 38.759 38.785C37.7973 39.7677 38.5528 39.0137 37.5567 39.9049C37.5338 39.9278 37.5109 39.9392 37.4994 39.9621C36.5148 40.9447 36.698 41.9044 36.9041 42.5557C36.9155 42.59 36.927 42.6243 36.9384 42.6586C37.7513 44.6238 38.8962 46.4748 40.6365 48.68L40.648 48.6914C43.808 52.5762 47.1397 55.604 50.8149 57.9235C51.2843 58.2206 51.7652 58.4605 52.2231 58.689C52.6353 58.8947 53.0246 59.0889 53.3566 59.2946C53.4024 59.3174 53.4482 59.3517 53.494 59.3746C53.8833 59.5688 54.2497 59.6602 54.6275 59.6602C55.5778 59.6602 56.1731 59.0661 56.3678 58.8718L58.612 56.6322C59.0012 56.2437 59.6195 55.7753 60.3408 55.7753C61.0507 55.7753 61.6346 56.2209 61.9895 56.6094C62.0009 56.6208 62.001 56.6208 62.0124 56.6322L68.3209 62.9279C69.5002 64.0933 68.3324 66.5042 68.3324 66.5042Z"
          stroke="#FEFCFD"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="6"
        />
      </svg>
      <a
        className="email"
        href="tel: 0134 271 8679"
      >
        0134 271 8679
      </a>
      <a
        className="home"
        href="/"
        onClick={[Function]}
      >
        Return to homepage
      </a>
    </div>
  </div>
</div>
`;
